src/AutoRest.CSharp/Mgmt/Output/ResourceData.cs

```diff
@@ -22,5 +23,14 @@ protected string CreateDescription(string clientPrefix)
         {
             return $"A class representing the {clientPrefix} data model.";
         }
+
+        internal bool IsIdString()
+        {
+            var baseTypes = EnumerateHierarchy().TakeLast(2).ToArray();
+            var baseType = baseTypes.Length == 1 || baseTypes[1].Declaration.Name == "Object" ? baseTypes[0] : baseTypes[1];
+            var idProperty = baseType.Properties.Where(p => p.Declaration.Name == "Id").First();
```

Comment:
Is it possible that property name will change in future?

Reply:
no this is an arm requirement